cohortextractor/date_expressions.py

```diff
@@ -323,7 +323,7 @@ def get_date_expression(self, date_column):
         date_format = date_column.date_format
         date_sql = re.sub(r"\s", "", str(date_column))
         match = re.match(
-            r"ISNULL\(CONVERT\(VARCHAR\(\d+\),([#\w\.]+),23\),''\)", date_sql
+            r"ISNULL\(CONVERT\(VARCHAR\(\d+\),([#\w\.\[\]]+),23\),''\)", date_sql
```

Comment:
Might save someone twenty seconds to give an example of exactly what we're matching?